In [1]:
import pandas as pd
import numpy as np

In [20]:
# Read the dataset into a pandas dataframe
df = pd.read_csv("green_tripdata_2020-01.csv", \
            dtype={'PULocationID': int, 'total_amount': float}, \
            usecols=['PULocationID', 'total_amount'])

In [21]:
df

PULocationID  total_amount
0                264          4.81
1                 66         24.36
2                181         15.34
3                129         25.05
4                210         11.30
...              ...           ...
447765           167         26.26
447766            81         30.32
447767           244         29.00
447768            68         33.44
447769            22         45.25

[447770 rows x 2 columns]

In [22]:
taxi_data = df.groupby(['PULocationID']).sum().reindex(range(1, 256)).fillna(0)

In [24]:
def myLog(x):
    return np.log(x) if (x > 0) else 0

#taxi_data['PULocationID'] = taxi_data.index
taxi_data.reset_index(level=0, inplace=True)
taxi_data["log_total_amount"] = np.array(list(map(myLog, taxi_data["total_amount"])))

In [25]:
taxi_data

PULocationID  total_amount  log_total_amount
0               1        466.61          6.145494
1               2          0.00          0.000000
2               3      14820.08          9.603738
3               4       2972.79          7.997256
4               5          0.00          0.000000
..            ...           ...               ...
250           251        220.07          5.393946
251           252       6680.33          8.806923
252           253        644.13          6.467901
253           254      40336.71         10.605017
254           255      69236.94         11.145290

[255 rows x 3 columns]

In [26]:
taxi_data.to_csv("taxi_data.csv", encoding='utf-8', index=False)

In [50]:
dfzone = pd.read_csv('taxi+_zone_lookup.csv')

In [51]:
dfzone

LocationID        Borough                     Zone service_zone
0             1            EWR           Newark Airport          EWR
1             2         Queens              Jamaica Bay    Boro Zone
2             3          Bronx  Allerton/Pelham Gardens    Boro Zone
3             4      Manhattan            Alphabet City  Yellow Zone
4             5  Staten Island            Arden Heights    Boro Zone
..          ...            ...                      ...          ...
260         261      Manhattan       World Trade Center  Yellow Zone
261         262      Manhattan           Yorkville East  Yellow Zone
262         263      Manhattan           Yorkville West  Yellow Zone
263         264        Unknown                       NV          NaN
264         265        Unknown                      NaN          NaN

[265 rows x 4 columns]

LocationID 264 and 265 can be dropped as they also don't have a corresponding polygon in the shapefile